pyqt-apps/siriushla/li_ap_mps/main.py

```diff
@@ -0,0 +1,533 @@
+''' Diagnostic Interface of the LINAC's BPM'''
+from qtpy.QtCore import Qt, QEvent
+from qtpy.QtWidgets import QWidget, QGroupBox, QHBoxLayout, \
+    QVBoxLayout, QGridLayout, QLabel, QTabWidget, \
+    QPushButton
+import qtawesome as qta
+from pydm.widgets import PyDMLabel, PyDMSpinbox
+from .util import PV_MPS, MPS_PREFIX, CTRL_TYPE, GROUP_POS, \
+    GROUP_POSALL, LBL_MPS, LBL_WATER, PV_TEMP_MPS, TEMP_TYPE
+from ..util import get_appropriate_color
+from ..widgets import SiriusMainWindow, PyDMLedMultiChannel,\
+     PyDMLed, SiriusPushButton
+from .bypass_btn import BypassBtn
+
+
+class MPSControl(QWidget):
+    ''' Monitor Protection System Controller Interface '''
+
+    def __init__(self, parent=None, prefix=''):
+        '''Contain all the graphic interface data'''
+        super().__init__(parent)
+
+        self.prefix = prefix + ('-' if prefix else '')
+        self.pv_obj = {}
+        self.led_clicked = False
+        self.all_clicked = False
+        self.stop = False
+
+        color = get_appropriate_color('LI')
+        self.setObjectName('LIApp')
+        self.setWindowIcon(qta.icon('mdi.monitor-dashboard', color=color))
+
+        self.setWindowTitle('LI MPS Controls')
+        self._setupUi()
+
+    def eventFilter(self, obj_wid, event):
+        ''' Hover listener with hover function'''
+        obj = self.pv_obj.get(obj_wid)
+        if event.type() == QEvent.Leave:
+            if self.led_clicked:
+                self.controlHiddenBox(
+                    obj.get("name"), obj.get("layout"), obj.get("config"))
+                self.stop = False
+        return False
+
+    def getDeviceName(self, pv_name):
+        ''' Set device name '''
+        if pv_name.find('LA-RF:LLRF:KLY') != -1:
+            device_name = ''
+        else:
+            device_name = MPS_PREFIX
+        return device_name
+
+    def getCtrlWidget(self, pv_name, ctrl_type, config):
+        '''Display the specific control widget'''
+        device_name = self.getDeviceName(pv_name)
+        if ctrl_type in ['_I', '_L', '']:
+            ch2vals = {
+                device_name + pv_name + ctrl_type: config}
+            widget = PyDMLedMultiChannel(self)
+            widget.set_channels2values(ch2vals)
+            if pv_name == 'HeartBeat':
+                widget.setOffColor(PyDMLed.Yellow)
+        elif ctrl_type == '_B':
+            widget = BypassBtn(
+                self,
+                init_channel=device_name + pv_name + ctrl_type)
+        elif ctrl_type == '_R':
+            widget = SiriusPushButton(
+                self,
+                init_channel=device_name + pv_name + ctrl_type,
+                label='Reset',
+                pressValue=1,
+                releaseValue=0)
+        widget.setStyleSheet('''
+            max-width: 50px;
+            max-height: 20px;
```

Comment:
I think it is better to use `em` size references, to avoid size issues when increasing font size. Maybe, to help with the column align issues, we can use minimum size also.
```
            min-width: 3.2em; max-width: 3.2em;
            min-height: 1.29em; max-height: 1.29em;